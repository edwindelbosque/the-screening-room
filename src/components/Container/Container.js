import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import './Container.scss';
import PropTypes from 'prop-types';
import SearchBar from '../SearchBar/SearchBar';

export const Container = ({ movies, updateFavorites, type }) => {
	const allMovies = movies.map((movie, index) => (
		<MovieCard
			key={index}
			type={type}
			movie={movie}
			updateFavorites={updateFavorites}
		/>
	));
	return (
		<main className='Container'>
			<SearchBar />
			{allMovies}
		</main>
	);
};

export default Container;

Container.propTypes = {
	movies: PropTypes.array,
	updateFavorites: PropTypes.func,
	type: PropTypes.string
};
