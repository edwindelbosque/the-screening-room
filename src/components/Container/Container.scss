.Container {
	display: grid;
	grid-gap: 50px 40px;
	grid-template-columns: repeat(auto-fit, minmax(330px, 1fr));
	grid-template-rows: auto;
	justify-items: center;
	padding: 40px 100px;
	margin-top: 40px;
}

@media only screen and (max-width: 902px) {
	.Container {
		grid-gap: 20px 0px;
		grid-template-columns: (300px, 500px);
		padding: 25px 0;
	}
}
