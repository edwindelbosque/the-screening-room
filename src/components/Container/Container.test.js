import React from 'react';
import { shallow } from 'enzyme';
import Container from './Container';

describe('Container', () => {
  it('should match the All Movies snapshot', () => {
    let movies = [{ movie_id: 1 }, { movie_id: 2 }, { movie_id: 3 }];
    let updateFavorites = jest.fn();
    let type = 'movies',
      allMoviesWrapper = shallow(
        <Container
          movies={movies}
          updateFavorites={updateFavorites}
          type={type}
        />
      );

    expect(allMoviesWrapper).toMatchSnapshot();
  });
});
