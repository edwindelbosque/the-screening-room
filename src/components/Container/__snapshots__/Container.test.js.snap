// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the All Movies snapshot 1`] = `
<main
  className="Container"
>
  <Connect(SearchBar) />
  <withRouter(Connect(MovieCard))
    movie={
      Object {
        "id": 1,
      }
    }
    type="movies"
    updateFavorites={[MockFunction]}
  />
  <withRouter(Connect(MovieCard))
    movie={
      Object {
        "id": 2,
      }
    }
    type="movies"
    updateFavorites={[MockFunction]}
  />
  <withRouter(Connect(MovieCard))
    movie={
      Object {
        "id": 3,
      }
    }
    type="movies"
    updateFavorites={[MockFunction]}
  />
</main>
`;
